File: 滑动窗口/定长滑动窗口/1984.py
from typing import List
from collections import defaultdict, Counter, deque
from itertools import permutations, combinations
from datetime import datetime, date, time, timedelta
from functools import cmp_to_key, lru_cache
from math import gcd, sqrt, log, ceil, floor
from bisect import bisect_left, bisect_right
from heapq import heappush, heappop
import sys
sys.setrecursionlimit(10 ** 6 + 1)

class Solution:
    # def minimumDifference(self, nums: List[int], k: int) -> int:
    #     n = len(nums)
    #     nums.sort()
    #     return min(nums[i + k - 1] - nums[i] for i in range(n - k + 1))
    
    def minimumDifference(self, nums: List[int], k: int) -> int:
        return min(i - j for i, j in zip(sorted(nums)[k - 1:], sorted(nums)[:-k + 1])) if k != 1 else 0

File: 每日一题/复健/1984.cpp
#include <bits/stdc++.h>
using namespace std;
const int MOD = 1e9 + 7;

class Solution {
public:
    int minimumDifference(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int n = nums.size();
        int ans = INT_MAX;
        for (int i = 0; i < n - k + 1; i++)
            ans = min(ans, nums[i + k - 1] - nums[i]);
        return ans;
    }
};